### DISCUSSION TITLE
(Suggestion) Attached Commands timer in between commands

### DISCUSSION THREAD

**Discussion by johnsoncui02** [Zrips/CMI#3463] [Enhancement]
**Description of issue or feature request:**
Allow an option to create items with a timer to execute multiple commands in different time.

Example: [command_1] !timer 60 [command_2] or something similar.

Purpose: it helps with certain plugins with manual events turned on and manually turning things off sucks.

**Comment by mrfloris** [comment]
I know a way around this perhaps if we can't use delay! inside attachcommand. 
You could consider making a cmi alias with cmi aliaseditor, and add a new one - set the permission to private, and add the commands. 

You can either use delay! 5 to add 5 seconds between command 1 and command 2 on the third line.. 
```
- cmi heal [playerName]
- delay! 5
- cmi heal [playerName] 
```

Or you could use `cooldown:5?#! cmi heal [playerName]` if you wnat the whole command to have a cooldown.

Then add the alias as a `asConsole! customaliasnamehere`

Not sure how ideal this is, and probably also have some caveats, but maybe for now, this potentially could solve it. It really depends on what exactly you wish to achieve.

**Comment by johnsoncui02** [comment]
It seems to have worked and does exactly what it needs to. Quite awkwardly delay! works in custom alias. If so then all we need to do is just make !delay a available function for other usages as well :).

**Comment by CloudeLecaw** [comment]
why dont create an item that is executing the custom alias?
1. create custom alias command with your delays
2. attach the custom alias to the item
